licensedb/analysis.go

```diff
@@ -0,0 +1,74 @@
+package licensedb
+
+import (
+	"net/url"
+	"os"
+	"sort"
+	"sync"
+
+	"gopkg.in/src-d/go-license-detector.v2/licensedb/filer"
+)
+
+// Analyse runs license analysis on each item in `args`
+func Analyse(args ...string) []Result {
```

Comment:
Using this function is gonna be quite hard because it will require going one by one in the results and check for an error.

I'm thinking we could instead return `([]map[string]float, error)` to keep it closer to what we have in `func Detect(filer Filer) (map[string]float, error)`

That way we don't need to expose new types into the API (`Match` and `Result`) and instead return a slice with the results of calling `Detect` on each one.

The error could be either a single error with the first error we encountered, or a new type containing a list of all of the errors we encountered (if any)

Reply:
@campoy I don't think returning a `([]map[string]float, error)` would work, as there are multiple licences matched for a single file tree and so you won't be able to tell which licences match belong to which file tree.
Returning a `map[filer.Filer]map[string]float32, error` would get your closer and give you this association, but then you would also lose the ability to return an error per file tree.

This is where the `Result` struct comes in, as it allows to store multiple licences for a file tree or an error otherwise. API users can either iterate through the `[]Result` or parse it as JSON.
Besides, as an API user I find it much easier to deal with a data structure that is clearly defined in a custom struct rather than a very generic structure like []map which is more opaque and so harder to reason about.

Reply:
I disagree with the logic behind returning `[]Error`, but not enough to block this.
LGTM